internal/gonut/cmd/push.go

```diff
@@ -202,13 +216,44 @@ func genericCommandFunc(cmd *cobra.Command, args []string) {
 }
 
 func runSampleAppPush(app sampleApp) error {
+
+	flags := []string{}
+
+	if len(buildpackSetting) > 0 {
+		app.buildpack = buildpackSetting
+		flags = append(flags, "-b", buildpackSetting)
+	}
+
+	app.stack = stackSetting
+	flags = append(flags, "-s", stackSetting)
+
+	hasStack, err := cf.HasStack(app.stack)
+	if err != nil {
+		return err
+	}
+
+	// Skip sample app push if desired stack is unavailable
+	if !hasStack {
+		bunt.Printf("Skipping push of *%s* sample app, because there is no DarkSeaGreen{%s} stack installed.\n",
+			app.caption,
+			app.buildpack,
+		)
+
+		return nil
```

Comment:
I think this should be an error case. If stack is set and not installed, we should fail.

Reply:
Do you mean it should return a non-zero exit code? In this case it will stop the execution of the command due to the return anyway. The same procedure for a missing buildpack because the err object is nil - The only thing is that it returns a 0 exit code which would be bad for automation.